### PR TITLE
Add specification for 'Add point' feature

### LINKED ISSUE #109: Add specification for adding a point
- [ ] Adding a point manually
- [x] Adding a point from GPS

### PR DESCRIPTION
**Pull request by ekigamba**
Fixes #109

### CODE REVIEW

**Comment by jasonrogena** [comment]
@ekigamba are we meant to allow for custom markers for the dropped points?

**Comment by ekigamba** [comment]
Not in this specification, would you be ok if I did any view customisations last?

**Comment by ekigamba** [comment]
> @ekigamba are we meant to allow for custom markers for the dropped points?

Moved to https://github.com/onaio/kujaku/issues/125

File: SPECIFICATION.md [hunk 1]
```diff
@@ -0,0 +1,441 @@
...
+   * [Kujaku Map View](#kujaku-map-view)
+   * [Helper Classes](#helper-classes)
+     * [MapBoxStyleHelper class](#1-mapboxstylehelper)
+     * [CoordinateUtils class](#2-coordinateutils)
+     * [MapBoxWebServiceApi](#3-mapboxwebserviceapi)
+
+
+## What is a Geospatial Widget?
+
+The Geospatial widget is an Android SDK designed to connect to the georegistry and other common geographical data sources like OSM and OGC servers. It also supports visualisation of geosptial data and inspection. It is designed to integrate into common mobile data collection tools used in global health.  It is expected to support the three elimination use cases including:
```
Comment (githengi) [inline SPECIFICATION.md]: Could you reword

>  It is expected to support the three elimination use cases including:

Maybe remove word elimination and three 


>  It is expected to support the below use cases among others:

**Comment by githengi** [comment]
Also add a link of the specification in the readme menu

**Comment by ekigamba** [comment]
Cool

File: SPECIFICATION.md (hunk 1 above)
Comment (ekigamba) [reply SPECIFICATION.md]: Cool, Reworded to `It is expected to support the following Malaria elimination use cases among others`

File: SPECIFICATION.md [hunk 2]
```diff
@@ -0,0 +1,441 @@
...
+
+```java
+
+    String PARCELABLE_KEY_MAPBOX_ACCESS_TOKEN;  // Mapbox API access token
+    String[] PARCELABLE_KEY_MAPBOX_STYLES;  // Mapbox Styles (https://www.mapbox.com/mapbox-gl-js/style-spec/)
+```
+
+The following is what should be sent in the `PARCELABLE_KEY_MAPBOX_STYLES`:
+
+Index 0 should have either of the following:
```
[the remark this answers is not in the document]
Comment (ekigamba) [reply SPECIFICATION.md]: index 1 and > do not have anything for now. I don't remember what the plan or vision was when using the array, might have been to be able to merge sources and layers inside the widget itself in a case where the style was a link

File: SPECIFICATION.md [hunk 3]
```diff
@@ -0,0 +1,441 @@
...
+
+For the `MapActivity` to respond to clicks on a feature, the feature requires to have:
+- Specified in the **Kujaku config**
+- Properties defined for it
+- An `id` as one of the `properties`
+
+
+## How to create a Mapbox style with Kujaku configuration
+
+The aim is to stick to a certain spec i.e. add the Kujaku configuration as close to the Mapbox specification. [Here](sample_mapbox_style_with_kujaku_config.json) is a sample style with the Kujaku configuration.
```
[the remark this answers is not in the document]
Comment (ekigamba) [reply SPECIFICATION.md]: 👍

File: SPECIFICATION.md [hunk 4]
```diff
@@ -0,0 +1,441 @@
...
+
+        startActivity(intent);
+```
+
+
+2. Start an activity with data
+
+Go [here](#mapboxstylehelper) for more on the how to create a mapbox style with your own geospatial data
+
+```java
```
[the remark this answers is not in the document]
Comment (ekigamba) [reply SPECIFICATION.md]: Fixed that

File: SPECIFICATION.md [hunk 5]
```diff
@@ -0,0 +1,441 @@
...
+        });
+        intent.putExtra(Constants.PARCELABLE_KEY_MAPBOX_ACCESS_TOKEN, "sdklcs823k9OIDFSKsd8uwk");
+
+        startActivity(intent);
+```
+
+
+2. Start an activity with data
+
+Go [here](#mapboxstylehelper) for more on the how to create a mapbox style with your own geospatial data
```
[the remark this answers is not in the document]
Comment (ekigamba) [reply SPECIFICATION.md]: 👍

**Pull request merged**